pulp_container/app/registry_api.py

```diff
@@ -947,9 +914,7 @@ def put(self, request, path, pk=None):
 
             config_digest = config_layer.get("digest")
             try:
-                config_blob = models.Blob.objects.get(
-                    digest=config_digest, pk__in=repository.latest_version().content
-                )
+                config_blob = models.Blob.objects.get(digest=config_digest, pk__in=content_pks)
```

Comment:
how about you change this to:
config_blob_qs =  models.Blob.objects.filter(digest=config_digest, pk__in=content_pks)
if not config_blob_qs.exists():
    raise BlobInvalid

and reuse this qs by extending found_blobs after line 993